### ISSUE TITLE
Epoch 51: early stopping

### ISSUE THREAD

**Issue by Zain-ul-abdin0** [pythonlessons/mltu#28]
Hi!
Epoch 51: early stopping
Having this error. Is it occur if we train with dataset less than thousand images? I am training with my 900 labled images given below but when I try to predict text it shows empty string. 
![2a4m2e](https://github.com/pythonlessons/mltu/assets/39799090/6ac8c679-09e5-4b87-949b-cfcc2e2d6131)
![2a5bpc](https://github.com/pythonlessons/mltu/assets/39799090/a1406607-a9fc-4948-b168-d583ebaf2fe6)
![2a8722](https://github.com/pythonlessons/mltu/assets/39799090/19dba5b3-b837-4be8-a526-8709292d6a52)
![2d8dny](https://github.com/pythonlessons/mltu/assets/39799090/7b37d14d-4835-49ea-9168-68dbba2cb74f)
![2de8g](https://github.com/pythonlessons/mltu/assets/39799090/79f8adb3-e4ad-469d-8bd1-80b03264f4d3)

![2a2gn2](https://github.com/pythonlessons/mltu/assets/39799090/6bc4f6d7-2624-41da-b141-5fc6c138f8d7)

**Comment by pythonlessons** [comment]
your code doesn't match image, not clear what is model

**Comment by Zain-ul-abdin0** [comment]
This trained model is not predicting my images sir.

**Comment by pythonlessons** [comment]
> This trained model is not predicting my images sir.

show your code, how you do this prediction, because I can't tell why without code...

**Comment by Zain-ul-abdin0** [comment]
when I run train.py, it stopped training after sometime and give this message Epoch 51: early stopping and after that when I try to predict image. it did not predict. Below 3 lines I added, it worked with different data set nut not with this.

import cv2
import typing
import numpy as np

from mltu.inferenceModel import OnnxInferenceModel
from mltu.utils.text_utils import ctc_decoder, get_cer

class ImageToWordModel(OnnxInferenceModel):
def init(self, char_list: typing.Union[str, list], *args, **kwargs):
super().init(*args, **kwargs)
self.char_list = char_list

def predict(self, image: np.ndarray):
    image = cv2.resize(image, self.input_shape[:2][::-1])

    image_pred = np.expand_dims(image, axis=0).astype(np.float32)

    preds = self.model.run(None, {self.input_name: image_pred})[0]

    text = ctc_decoder(preds, self.char_list)[0]

    return text
if name == "main":
import pandas as pd
from tqdm import tqdm
from mltu.configs import BaseModelConfigs

configs = BaseModelConfigs.load("./Models/02_captcha_to_text/202308060003/configs.yaml")
model = ImageToWordModel(model_path=configs.model_path, char_list=configs.vocab)

df = pd.read_csv("./Models/02_captcha_to_text/202308060003/val.csv").values.tolist()

accum_cer = []
# for image_path, label in tqdm(df):
#     image = cv2.imread(image_path)

#     prediction_text = model.predict(image)

#     cer = get_cer(prediction_text, label)
#     print(f"Image: {image_path}, Label: {label}, Prediction: {prediction_text}, CER: {cer}")

#     accum_cer.append(cer)

image = cv2.imread('./6wf4ef.jpg')
prediction_text = model.predict(image)
print(f"Predicted Text: {prediction_text}")

**Comment by pythonlessons** [comment]
what was your CER while training model on validation data, I think your model was not trained

**Comment by pythonlessons** [comment]
Yes, I can train you a model for $$. Otherwise, I can't train a model for everyone. If you are interested write me an email to pythonlessons0@gmail.com

**Comment by zainulabdinsystems** [comment]
I can give you $$ but how much? I need to crack all images of this type.
![](https://user-images.githubusercontent.com/39799090/259132553-c0c28ad8-dd06-45e6-a8bb-5ac40f74b09a.jpg)

**Comment by ghost** [comment]
hello, were u able to eventually fix this?
